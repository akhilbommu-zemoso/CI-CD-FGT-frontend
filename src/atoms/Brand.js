import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import { Link} from 'react-router-dom'
const useStyles = makeStyles((theme) => ({
    title: {
        height: '30px',
        fill: 'white',
        borderRadius: "50%",
        padding: '5px',
        marginBottom: '8px',
        marginRight: '10px',
        fontSize: '28px',
        fontWeight: 900,
        color:'#485460',
    },
    linksclass: {
        color: 'black',
        paddingTop: '0px',
        textDecoration: 'none',
        '&:hover': {
            color: 'black',
            textDecoration: 'none',
        },
    },
}));
export default function Brand(props) {
    const classes = useStyles();
    return (
        <div>
            <List component="nav" aria-label="main mailbox folders">
                <Link className={classes.linksclass} to="/dashboard/home">
                    <div className={classes.title} id="brand">
                        {props.name}
                     </div>
                </Link>
            </List>
        </div>
    );
}
