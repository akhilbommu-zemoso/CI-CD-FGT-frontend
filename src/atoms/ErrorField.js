import React from 'react'
import { makeStyles } from '@material-ui/core/styles';
const useStyles = makeStyles({
    error:{
        color:'red',
        fontSize: "12px",
        marginLeft: '10px',
        height: '10px'
    }
});

function ErrorField(props) {
    const classes = useStyles();
    return (
        <div className={classes.error}>
            {props.message}
        </div>
    )
}
export default ErrorField