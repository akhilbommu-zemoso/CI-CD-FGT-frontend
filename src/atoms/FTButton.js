import React from 'react'
import Button from '@material-ui/core/Button'
import { makeStyles } from '@material-ui/core/styles';
const useStyles = makeStyles({
    color: {
        color: 'white',
        textTransform: 'none',
        backgroundColor: 'rgb(4,136,255)',
        '&:hover': {
            color: 'white',
            backgroundColor: 'rgb(4,136,255)'
        },
    }
});

function FTButton(props) {
    const classes = useStyles();
    return (
        <Button
                id="button"
                data-testid="button"
                variant="contained"
                classes={{ root: classes.color }}
                onClick={props.method}
                disabled={props.disabled}
        >
            {props.name}
        </Button>
    )
}
export default FTButton