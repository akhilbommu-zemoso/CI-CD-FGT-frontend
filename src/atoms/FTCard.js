import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";
import { Divider } from "@material-ui/core";
function FTCard(props) {
  return (
    <Card className={props.cardStyle}>
      <CardContent>
        <Typography className={props.headerStyle}>{props.header}</Typography>
        <Typography className={props.contentStyle}>{props.content}</Typography>
        <Typography className={props.footerStyle}>{props.footer}</Typography>
      </CardContent>
    </Card>
  );
}
function CardHeader(props) {
  return (
    <div>
      <div className={props.headerStyle}>
        <Typography className={props.headerTitleStyle}>
          {props.headerTitleContent}
        </Typography>
        <div>{props.headerRight}</div>
      </div>
      <Divider style={{ marginTop: '10px' }} />
    </div>
  );
}
export { FTCard, CardHeader };
