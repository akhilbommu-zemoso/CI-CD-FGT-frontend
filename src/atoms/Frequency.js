import React from 'react'
import { makeStyles } from '@material-ui/core/styles';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormHelperText from '@material-ui/core/FormHelperText';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';
function Frequency(props) {
    const useStyles = makeStyles((theme) => ({
        formControl: {
            fontSize: '14px',
            '&:focus': {
                outline: 'none',
                border: 'none',
                textDecoration: 'none'
            },
        },
        selectEmpty: {
            width: 130,
            height: 35,
            maxHeight: '50px',
            marginLeft: 12,
            fontSize: '14px',

        },
        top: {

        }
    }));
    const classes = useStyles();
    return (
        <div>
            <FormControl variant="outlined" className={classes.formControl}>
                <div>
                    <Select
                        data-testid="select"
                        labelId="demo-simple-select-outlined-label"
                        id="demo-simple-select-outlined"
                        name={props.name}
                        value={props.value}
                        onChange={(event) => props.method(event)}
                        displayEmpty
                        className={classes.selectEmpty}
                        inputProps={{ 'aria-label': 'Without label' }}
                    >
                       { props.items.map((item,index)=>{
                            return <MenuItem value={item}>{item}</MenuItem>
                        })}
                    </Select>
                </div>
            </FormControl>
        </div>
    )
}

export default Frequency
