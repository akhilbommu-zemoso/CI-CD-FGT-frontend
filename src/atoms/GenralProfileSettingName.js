import React from 'react'
import Grid from '@material-ui/core/Grid';
import { makeStyles } from '@material-ui/core/styles';
const useStyles = makeStyles((theme) => ({
    top: {
        marginTop: 5,
        marginBottom: 20,
        fontSize: '14px',
    },
}));
function GenralProfileSettings(props) {
    const classes = useStyles();
    return (
        <Grid item xs={6}>
            <div className={classes.top}>
                {props.setting}
            </div>
        </Grid>
    )
}
export default  GenralProfileSettings 
