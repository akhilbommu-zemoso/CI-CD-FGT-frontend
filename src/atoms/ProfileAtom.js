import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import { Link, Redirect } from 'react-router-dom'
import { logout } from '../utils/index'
import ProfilePicture from '../components/ProfilePicture';
const useStyles = makeStyles((theme) => ({
    circle: {
        width: 60,
        height: 60,
        marginRight: 20,
        backgroundColor: "rgba(182,32,224,0.71)",
        fontSize: 28,
      },
    newline: {
        display: 'flex',
        flexDirection: 'column',
        fontSize: '14px',

    },
    linksclass: {
        color: 'black',
        paddingTop: '0px',
        fontSize: '14px',
        textDecoration: 'none',
        '&:hover': {
            color: 'black',
            textDecoration: 'none',
        },
    },
    margin: {
        marginTop: 15,
        fontSize: '14px',
    },
    font: {
        fontSize: '14px',
    }
}));
function ProfileHeader() {
    const user = JSON.parse(localStorage.getItem("user"));
    const classes = useStyles();
    return (
        <div>
            <List component="nav" aria-label="main mailbox folders">
                <Link to="/profile" className={classes.linksclass}>
                    <ListItem button>
                        <ListItemIcon>
                            <ProfilePicture pictureStyle={classes.circle}/>
                        </ListItemIcon>
                        <div className={classes.newline}>
                            <ListItemText classes={{ primary: classes.font }} primary={user.name} />
                            <ListItemText classes={{ primary: classes.font }} primary={user.email} />
                        </div>
                    </ListItem>
                </Link>
            </List>
        </div>
    );
}
function ProfileFooter() {
    const classes = useStyles();

    const logoutUser = () => {
        logout();
        return <Redirect to="/" />
    }
    return (
        <div>
            <Link data-testid="link" className={classes.linksclass} onClick={logoutUser}>
                <div className={classes.margin} >
                    <ListItem button>
                        <ListItemText classes={{ primary: classes.font }} primary="Logout" />
                    </ListItem>
                </div>
            </Link>
        </div>
    );
}
export { ProfileHeader, ProfileFooter }