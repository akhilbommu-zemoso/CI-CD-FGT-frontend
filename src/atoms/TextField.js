import React from 'react'
import { makeStyles, withStyles, fade } from "@material-ui/core/styles";
import { InputBase } from "@material-ui/core"
const BootstrapInput = withStyles((theme) => ({
    root: {
        "label + &": {
            marginTop: theme.spacing(3),
        },
        fontSize: '14px',
    },
    margin: {
        margin: theme.spacing(1),
    },
    input: {
        borderRadius: 4,
        position: "relative",
        backgroundColor: "white",
        border: "1px solid #ced4da",
        fontSize: '14px',
        width: "300px",
        height: "27px",
        padding: "6px 12px",
        transition: theme.transitions.create(["border-color", "box-shadow"]),
        "&:focus": {
            boxShadow: `${fade(theme.palette.primary.main, 0.25)} 0 0 0 0.2rem`,
            borderColor: theme.palette.primary.main,
        },
    },
}))(InputBase);
export default BootstrapInput