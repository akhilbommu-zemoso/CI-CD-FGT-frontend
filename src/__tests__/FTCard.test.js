import React, { Component } from 'react'
import { render, fireEvent, getAllByText, screen } from '@testing-library/react'
import { FTCard,CardHeader } from '../atoms/FTCard'

it('should take a snapshot', () => {
    const { asFragment} = render(<FTCard/>)
    expect(asFragment(<FTCard/>)).toMatchSnapshot()
});
it('should take a snapshot', () => {
    const { asFragment} = render(<CardHeader  />)
    expect(asFragment(<CardHeader/>)).toMatchSnapshot()

});

it('checks for FTCard header content footer through props', () => { 
    const { getbyText } = render(<FTCard header={"Iam header"} footer={"Iam footer"} content={"Iam content"} />)  
    expect(screen.getByText("Iam header")).toBeInTheDocument
    expect(screen.getByText("Iam footer")).toBeInTheDocument
    expect(screen.getByText("Iam content")).toBeInTheDocument
})

it('checks CardHeader content through props', () => {
    const { getByText } = render(<CardHeader headerTitleContent="Card header" />)
    expect(screen.getByText("Card header")).toBeInTheDocument
   
})