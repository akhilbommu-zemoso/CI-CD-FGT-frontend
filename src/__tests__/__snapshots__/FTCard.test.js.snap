// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      />
      <p
        class="MuiTypography-root MuiTypography-body1"
      />
      <p
        class="MuiTypography-root MuiTypography-body1"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`should take a snapshot 2`] = `
<DocumentFragment>
  <div>
    <div>
      <p
        class="MuiTypography-root MuiTypography-body1"
      />
      <div />
    </div>
    <hr
      class="MuiDivider-root"
      style="margin-top: 10px;"
    />
  </div>
</DocumentFragment>
`;
